### DISCUSSION TITLE
Proper weak form

### DISCUSSION THREAD

**Discussion by Leni-Yeo** [idaholab/moose#21937] [Q&A Modules: Phase field] [answered]
Good morning, 
I am trying to figure out what the weak form from the last part (kernel) in "New phase-field model for polycrystalline systems with anisotropic grain boundary properties" from "Nele Moelans" would look like: -m▽ .  (∑_(j≠i) ((∂γ_ij)/(∂▽η_i )) η_i^2  η_j^2)

![12](https://user-images.githubusercontent.com/78172130/186726476-ac4b977b-afea-4313-aca3-36b1ef523f8a.PNG)

I think the weak form should be : ( ( (-1) * m * 2 * η_i * grad(η_i) * (∂γ_ij)/(∂▽η_i ) * ∑_(j≠i)  η_j^2 ) , ψ )
I am taking everything outside the gradient as constants and doing ▽ . η_i^2 =  η_i * grad(η_i)  +  η_i * grad(η_i) =  2 * η_i * grad(η_i)
 
I am not sure if I am correct on that. If anyone has some insight, please advise since I am relatively new to phase field.

Thank you in advance.

**Comment by GiudGiud** [comment, marked as answer]
Hello

the derivation of the weak form looks too short. There seems to be many more terms in your base equation than in your result

Feel free to paste a step by step derivation and we could point you to potential issues

Guillaume

**Reply by Leni-Yeo** [reply]
Good afternoon,
Thank you for the reply and sorry for the delay in response.
I redid the derivation and corrected a mistake I  found. I attached the new derivation. Please give your inputs on it whenever you can.
Thank you in advance.!

[Last Kernel Derivation.pdf](https://github.com/idaholab/moose/files/9472673/Last.Kernel.Derivation.pdf)

**Reply by GiudGiud** [reply]
I dont see the grad (eta_i^2) in your initial equation? I see grad(eta_i^2 eta_j^2)? 

also why is there only one term in your derivation?
Is this the only term you care about?

**Reply by Leni-Yeo** [reply]
I assumed the summation of eta_j^2 and the fraction (partial derivative) to come out from under the gradient operator as constants so that the the gradient operator only applies to eta_i^2. I thought that would be the case since only the ith order parameter is considered each time. I might be wrong there. ???

**Reply by GiudGiud** [reply]
is eta_j not a variable?
there are off-diagonal terms as well in the Jacobian, so eta_j being in the equation for eta_i will mean some of these contributions

**Comment by Leni-Yeo** [comment]
I see. I guess I will try again. Please advise whenever you can. Thank you!

**Reply by GiudGiud** [reply]
I recommend you lean hard on the examples in the phase field module. Try to identify models that are similar to what you are doing.

Also you should just leverage automatic differentiation if you can. Then you wont have to worry about coding up a jacobian, just the residual

**Comment by Leni-Yeo** [comment]
Will do. Thank you!
